CONTRIBUTING.md

```diff
@@ -1,18 +1,149 @@
-# Contributing
+# Contributing to ICU4X Project
+
+`ICU4X` is an open source project and welcomes everyone to participate.
+
+In order to provide meaningful contributions, it is important to familiarize yourself with a set of documents which describe the [structure](./docs/charter.md) and [style](./docs/style-guide.md) used by the project.
+
+The list of [open issues](triaging.md) represent the current focus points of the project, and the [help wanted](https://github.com/unicode-org/icu4x/issues?q=is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee+) lists the issues which we don't have resources to work on right now, but would consider accepting if someone volunteered to work on them.
+
+Issues are open to everyone to discuss and can be used to jump-start Pull Requests intended for the project.
+
+In most cases, the first step is to find or file a new issue related to your planned contribution, discuss it, and once you received a feedback indicating that the pull request would be welcomed you can start working on it.
+
+## Contributing a Pull Request
+
+The first step is to fork the repository to your namespace and create a branch off of the `master` branch to work with.
+
+That branch may end up containing one of more commits that are constituting the full scope of the pull request.
+
+### Checklist
+
+Each commit and pull request should follow the [style guide](./docs/style-guide.md) and pass all checks such as tests and `cargo fmt`. If the PR is adding any public API changes, we'd also like to ensure that full coverage of `cargo doc` is preserved and code coverage hovers above `90%`.
+
+### Structure of commits in a Pull Request
+
+Pull Request lifecycle is divided into two phases.
+
+The first one is the work done to get the Pull Request ready for review.
+The other is the review cycle.
+
+#### Draft Phase
+
+If the pull request is simple and short lived, it can be initialized with review request.
+If the pull request is more complex and is being developed over time, it may be benefitial to start it in a `Draft` state.
+This allows other contributors to monitor the progress and volunteer feedback while annotating that the pull request is not yet ready for review.
+
+By the end of this phase, and right before review is requested, it is helpful for the reviewers to have a clean list of commits in the pull request.
+
+In most cases, a single commit per pull request is enough.
+
+Multiple commits should be used when the commit is too large and the scope of changes can be reduced by separating it into multiple commits which are logically self-contained.
+Such commits do not have to pass tests in isolation, and need only to be meaningfully complete for the reviewer to benefit from reading, compared to reviewing all the changes at once.
+
+### Review Phase
+
+Once the pull request is ready, the author can switch from draft to regular pull request.
```

Comment:
Based on a sentence from the "Draft Phase" section ("Such commits do not have to pass tests in isolation..."), I think here, "ready" implies that the PR should be passing all tests before review.  If so, we might as well be specific about it.